quickwit/quickwit-serve/src/search_api/rest_handler.rs

```diff
@@ -229,31 +230,51 @@ async fn search_endpoint(
 }
 
 fn search_get_filter(
-) -> impl Filter<Extract = (String, SearchRequestQueryString), Error = Rejection> + Clone {
+) -> impl Filter<Extract = (Vec<String>, SearchRequestQueryString), Error = Rejection> + Clone {
     warp::path!(String / "search")
+        .and_then(extract_index_id_patterns)
         .and(warp::get())
         .and(serde_qs::warp::query(serde_qs::Config::default()))
 }
 
 fn search_post_filter(
-) -> impl Filter<Extract = (String, SearchRequestQueryString), Error = Rejection> + Clone {
+) -> impl Filter<Extract = (Vec<String>, SearchRequestQueryString), Error = Rejection> + Clone {
     warp::path!(String / "search")
+        .and_then(extract_index_id_patterns)
         .and(warp::post())
         .and(warp::body::content_length_limit(1024 * 1024))
         .and(warp::body::json())
 }
 
 async fn search(
-    index_id: String,
+    index_id_patterns: Vec<String>,
     search_request: SearchRequestQueryString,
     search_service: Arc<dyn SearchService>,
 ) -> impl warp::Reply {
-    info!(index_id = %index_id, request =? search_request, "search");
+    info!(request =? search_request, "search");
     let body_format = search_request.format;
-    let result = search_endpoint(index_id, search_request, &*search_service).await;
+    let result = search_endpoint(index_id_patterns, search_request, &*search_service).await;
     make_json_api_response(result, body_format)
 }
 
+pub(crate) async fn extract_index_id_patterns(
+    comma_separated_index_patterns: String,
+) -> Result<Vec<String>, Rejection> {
+    let mut index_ids_patterns = Vec::new();
+    for index_id_pattern in comma_separated_index_patterns.split(',') {
+        validate_index_id_pattern(index_id_pattern).map_err(|error| {
+            warp::reject::custom(crate::rest::InvalidArgument(error.to_string()))
+        })?;
+        index_ids_patterns.push(index_id_pattern.to_string());
+    }
+    if index_ids_patterns.is_empty() {
+        return Err(warp::reject::custom(crate::rest::InvalidArgument(
+            "Missing index ID.".to_string(),
+        )));
+    }
```

Comment:
your error handling code can never be triggered.
```suggestion
    assert!(index_ids_patterns.is_empty());
```